### PR TITLE
Improvements to surfaces, time travel code, and an attempt to fix YYC.

### PR DESCRIPTION
**Pull request by nkrapivin**
What's new:
- set windows exe version to 1.1.0
- added vsync menu option to video settings
- some objects called surface_reset_target while no surface has been set, GMS1 doesn't care, but GMS2 throws an error "Reset target when no target was set", then "UNBALANCED SURFACE STACK", and then it quits.
- also YoYo doesn't recommend creating surfaces in the Create event, rather you should only create a surface right when you need it. (https://docs.yoyogames.com/source/dadiospice/002_reference/surfaces/index.html docs actually say that you should create them in the Draw Event but that's not true)
- enabled "Create textures on Demand" in Windows config because the Windows Runner preloads all texture pages in memory, and draw_texture_flush does nothing (https://docs.yoyogames.com/source/dadiospice/002_reference/drawing/draw_texture_flush.html), but if you tick that checkbox it will only load texture pages when needed like on every other platform. (This is on by default in GMS2 by the way)
- time travel scripts never call shader_set if tts_ok is false, which will prevent the game from crashing.

TL;DR: add vsync to menu, fix errors that happen when importing to GMS2.

### CODE REVIEW

**Comment by nkrapivin** [comment]
Since Alex merged #67 a little bit too early (my bad), I'll finish off the translation PR here as a separate commit.

**Comment by VectorSatyr** [comment]
@AlexKhayrullin @nkrapivin Please determine if the switch to "Create Textures on Demands" creates a significant enough improvement for deployment to Android.

**Review by VectorSatyr — CHANGES_REQUESTED** [review]
- VSync is fine. May revisit this later.
- There appears to be irrelevant changes to the SttAndroidGamepad extension. Please investigate.
- _UNBALANCED SURFACE STACK_ issue is correct, but it doesn't necessarily need addressing in this repository since it is focused on development of the project in GameMaker: Studio, not GameMaker Studio 2. Up to you whether the relevant changes should be kept or omitted, since they do not break execution in GMS1.
- GameMaker Studio documentation only advises that surfaces are _volatile_ and may easily be lost outside of the executable's control. It makes no difference if surfaces are pre-loaded in the create event or in the draw event, so long as the draw code is properly guarding against the possibility that the surface may not always exist and should then be re-created. It _does_ recommend that all drawing should happen in the draw event and that any drawing activity should be kept to a minimum. Since surfaces are volatile, I would _strongly consider_ putting the surface creation and rendering right before drawing in the Draw Event to maximize the chance that will exist when it is needed.

**Comment by nkrapivin** [comment]
- There appears to be irrelevant changes to the SttAndroidGamepad extension. Please investigate.

They are relevant, I unticked all checkboxes except Android and Android (YYC) which resolved "Bad Image" window popping up when running the game. (Runner tried to load .ext file as a dll)

**Comment by nkrapivin** [comment]
- Please determine if the switch to "Create Textures on Demands" creates a significant enough improvement for deployment to Android.

It *is* enabled by default on every other platform except Windows, and you can't disable this afaik.

**Comment by nkrapivin** [comment]
> * There appears to be irrelevant changes to the SttAndroidGamepad extension. Please investigate.
> 
> They are relevant, I unticked all checkboxes except Android and Android (YYC) which resolved "Bad Image" window popping up when running the game. (Runner tried to load .ext file as a dll)

image

**Comment by VectorSatyr** [comment]
> 
> 
>     * There appears to be irrelevant changes to the SttAndroidGamepad extension. Please investigate.
> 
> 
> They are relevant, I unticked all checkboxes except Android and Android (YYC) which resolved "Bad Image" window popping up when running the game. (Runner tried to load .ext file as a dll)

Then it belongs in a separate commit. This is for graphical improvements, not bug fixes. Mixing concerns makes changes harder to track. People looking at the commit history won't immediately know that this also _so happens_ to fix a crash on Android platforms, because they wouldn't think it had anything to do with a graphical update.

> 
> 
>     * Please determine if the switch to "Create Textures on Demands" creates a significant enough improvement for deployment to Android.
> 
> 
> It _is_ enabled by default on every other platform except Windows, and you can't disable this afaik.

That, in and of itself, is not evidence of a performance improvement. The truth is, this option was considered before, but was ultimately deemed unnecessary, so there needs to be some tangible benchmarks if we are going to reconsider.

**Comment by nkrapivin** [comment]
> Then it belongs in a separate commit. This is for "Improvements to graphics stuff", not bug fixes to gamepads. Mixing concerns makes changes harder to track. People looking at the commit history won't immediately know that this also _so happens_ to fix a crash on Android platforms, because they wouldn't think it had anything to do with a graphical update.

It's fixing seven error windows on non-Android platforms, not an Android crash, because the engine is trying to load .ext file as a .dll.

Also do I make a separate commit then?

**Comment by VectorSatyr** [comment]
> 
> 
> > Then it belongs in a separate commit. This is for "Improvements to graphics stuff", not bug fixes to gamepads. Mixing concerns makes changes harder to track. People looking at the commit history won't immediately know that this also _so happens_ to fix a crash on Android platforms, because they wouldn't think it had anything to do with a graphical update.
> 
> It's fixing seven error windows on non-Android platforms, not an Android crash, because the engine is trying to load .ext file as a .dll.
> 
> Also do I make a separate commit then?

Yes, please.

**Comment by nkrapivin** [comment]
As a separate PR or revert this commit and force-push new ones?

**Comment by VectorSatyr** [comment]
Can you not add the additional commit to the merge request?

**Comment by nkrapivin** [comment]
That duplicates the changes of the previous one?

**Comment by nkrapivin** [comment]
Okay I've split this PR into multiple commits.

**Review by VectorSatyr — APPROVED** [review]

**Comment by nkrapivin** [comment]
I'll finish off translation stuff in a separate PR... when I'll finally get at least one Russian font that I can play with...
